plugins/lsp/conform-nvim.nix

```diff
@@ -36,97 +86,159 @@ in
             };
         ```
       '';
-    };
 
-    formatOnSave = helpers.defaultNullOpts.mkNullable' {
-      type =
-        with helpers.nixvimTypes;
-        either strLuaFn (submodule {
-          options = {
-            lspFallback = mkOption {
-              type = types.bool;
-              default = true;
-              description = "See :help conform.format for details.";
-            };
-            timeoutMs = mkOption {
-              type = types.int;
-              default = 500;
-              description = "See :help conform.format for details.";
-            };
-          };
-        });
-      # Unknown plugin default
-      description = ''
-        If this is set, Conform will run the formatter on save.
-        It will pass the table to conform.format().
-        This can also be a function that returns the table.
-        See :help conform.format for details.
+      format_on_save =
+        defaultNullOpts.mkNullable
+          (
+            with types;
+            either strLuaFn (submodule {
+              freeformType = types.attrsOf anything;
+              options = {
+                inherit
+                  lsp_format
+                  timeout_ms
+                  quiet
+                  stop_after_first
+                  ;
+              };
+            })
+          )
+          { }
+          ''
+            If this is set, Conform will run the formatter on save.
+            It will pass the table to `conform.format()`.
+            This can also be a function that returns the table.
+            See `:help conform.format` for details.
+          '';
+
+      default_format_opts =
+        defaultNullOpts.mkNullable
+          (
+            with types;
+            either strLuaFn (submodule {
+              freeformType = types.attrsOf anything;
+              options = {
+                inherit
+                  lsp_format
+                  timeout_ms
+                  quiet
+                  stop_after_first
+                  ;
+              };
+            })
```

Comment:
It seems this type is common to a few options? Perhaps the submodule type, or even the whole `either` type, should be defined in a let block somewhere? Unless I'm missing some subtle differences between them.

Reply:
They are just a few options that are shared between a few of the properties. There are a lot more supported options but these ones are the most likely to be used.